CRM/Member/BAO/Membership.php

```diff
@@ -139,38 +139,65 @@ public static function add(&$params, $ids = array()) {
     // reset the group contact cache since smart groups might be affected due to this
     CRM_Contact_BAO_GroupContactCache::opportunisticCacheFlush();
 
+    $activityParams = array();
+    $allStatus = CRM_Member_BAO_Membership::buildOptions('status_id', 'get');
+    if (!empty($params['is_pay_later']) ||
```

Comment:
Working through this function I think the question is when would we want to create the activity and for it NOT to be pending/ scheduled. It seems like the old behaviour was for it to always be created as Scheduled, when created. So, why don't we add a parameter 'membership_activity_status_id' that can be set to Completed if desired, but we default to scheduled? That way we aren't trying to deal with the BAO function deciding how pay later it is. And, the core form functions should always create pending and then call complete transaction, unless there is no payment at all I think. Getting away from is_pay_later & making the calling function specify the activity_status id seems more explicit & less random.

My other thinking on this was that the place where we REALLY know what is going on is in the MembershipPayment::create function. But I can't quite see how to turn that nugget into a strategy